File: android/src/main/java/dev/barooni/capacitor/calendar/models/inputs/ListEventsInRangeInput.kt
package dev.barooni.capacitor.calendar.models.inputs

import com.getcapacitor.PluginCall
import dev.barooni.capacitor.calendar.PluginError
import dev.barooni.capacitor.calendar.utils.ImplementationHelper

data class ListEventsInRangeInput(
    private var call: PluginCall,
) {
    val from: Long =
        call.getLong("from")?.let { ImplementationHelper.getCalendarFromTimestamp(it).timeInMillis } ?: throw PluginError.FromDateMissing
    val to: Long =
        call.getLong("to")?.let { ImplementationHelper.getCalendarFromTimestamp(it).timeInMillis } ?: throw PluginError.ToDateMissing
}

File: ios/Sources/CapacitorCalendarPlugin/Models/Inputs/ListEventsInRangeInput.swift
import Capacitor

struct ListEventsInRangeInput {
    private var from: Double
    private var to: Double

    init(call: CAPPluginCall) throws {
        guard let from = call.getDouble("from") else {
            throw PluginError.fromDateMissing
        }
        self.from = from
        guard let to = call.getDouble("to") else {
            throw PluginError.toDateMissing
        }
        self.to = to
    }

    func getFrom() -> Date {
        return ImplementationHelper.dateFromTimestamp(from)
    }

    func getTo() -> Date {
        return ImplementationHelper.dateFromTimestamp(to)
    }
}
